src/types/union.ts

```diff
@@ -70,10 +67,18 @@ export function Union<T extends readonly [RuntypeBase, ...RuntypeBase[]]>(
       }
     }
 
-    for (const targetType of alternatives)
-      if (innerValidate(targetType, value, visited).success) return SUCCESS(value);
+    return validate();
 
-    return FAILURE.TYPE_INCORRECT(self, value);
+    function validate() {
+      const details: Array<Details | string> = [];
+      for (const alternative of alternatives) {
+        const validation = innerValidate(alternative, value, visited);
+        if (validation.success) return SUCCESS(value);
+        else if (validation.details) details.push(validation.details);
+        else details.push(validation.message);
+      }
+      return FAILURE.CONTENT_INCORRECT(self, details);
```

Comment:
Hi. What if we split these changes into 2 PRs?

PR1: return `code: TYPE_INCORRECT` but with details
PR2: change `code` to `CONTENT_INCORRECT`

In this case PR1 does not have breaking changed, it can be merged in minor release

```suggestion
      const fail = FAILURE.CONTENT_INCORRECT(self, details);
      // TODO: remove this code hack in next major release
      fail.code = Failcode.TYPE_INCORRECT;
      return fail;
```

Reply:
@yuhr what do you think?
would you merge this PR if it preserves `code: TYPE_INCORRECT` but still adding details?

Reply:
Hm, sounds neat but how we should format the error message might differ between `TYPE_INCORRECT` and `CONTENT_INCORRECT` (not sure but probably), so let me proceed #296 before taking action on this. Thanks suggestion!

Reply:
Is this something that can be revisited? As more versions are published, it gets harder to keep up-to-date using a fork in a git reference.